packages/modal-ui-js/src/lib/modal.ts

```diff
@@ -0,0 +1,112 @@
+import type {
+  ModuleState,
+  Wallet,
+  WalletSelector,
+} from "@near-wallet-selector/core";
+import { renderModal } from "./render-modal";
+import type {
+  WalletSelectorModal,
+  ModalOptions,
+  ModalRoute,
+} from "./modal.types";
+import { renderWhatIsAWallet } from "./components/WhatIsAWallet";
+import { renderWalletAccount } from "./components/WalletAccount";
+
+const MODAL_ELEMENT_ID = "near-wallet-selector-modal";
+export const DEFAULT_DERIVATION_PATH = "44'/397'/0'/0'/1'";
+
+let modalInstance: WalletSelectorModal | null = null;
+
+type ModalState = {
+  container: HTMLElement;
+  selector: WalletSelector;
+  options: ModalOptions;
+  route: ModalRoute;
+  modules: Array<ModuleState<Wallet>>;
+  derivationPath: string;
+};
+
+export let modalState: ModalState | null = null;
+
+export function updateModalState(newModalState: ModalState) {
+  modalState = newModalState;
+}
+
+const el = document.createElement("div");
+el.id = MODAL_ELEMENT_ID;
+document.body.appendChild(el);
```

Comment:
We will need to wrap these three lines in an if statement to check if the window is defined otherwise it will fail in SSR frameworks. 

```ts
if (typeof window !== "undefined") {
  const el = document.createElement("div");
  el.id = MODAL_ELEMENT_ID;
  document.body.appendChild(el);
}
```